src/group_theory/submonoid/operations.lean

```diff
@@ -399,6 +399,15 @@ instance has_one : has_one S := ⟨⟨_, S.one_mem⟩⟩
 @[to_additive] lemma mul_def (x y : S) : x * y = ⟨x * y, S.mul_mem x.2 y.2⟩ := rfl
 @[to_additive] lemma one_def : (1 : S) = ⟨1, S.one_mem⟩ := rfl
 
+/-- A monoid is isomorphic to itself as a submonoid. -/
+@[to_additive "An additive monoid is isomorphic to itself as an additive submonoid."]
+def equiv_top : M ≃* (⊤ : submonoid M) :=
```

Comment:
This is frome #12658 right? Can you mark this PR as a dependent of #12658 in that case

Reply:
I've added it for you

Reply:
Thanks - I didn't know we could do this.